### PR TITLE
[Impeller] Allow metal shaders to compile through SPIR-V with openGL semantics.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/123409

Using `float16sampler2d` requires compiling through openGL semantics. We shouldn't do this by default because it prevents us from using the framebuffer fetch extension.

### CODE REVIEW

File: impeller/tools/impeller.gni [hunk 1]
```diff
@@ -666,12 +676,26 @@ template("_impeller_shaders_vk") {
 #
 #    The SPIR-V version required by the shaders.
 #
+# @param[options] use_half_textures
+#
+#    Whether the metal shader is using half-precision textures and requires
```
Comment (chinmaygarde) [inline impeller/tools/impeller.gni:681]: Can this be the default for Metal? Or is this because it interferes with the other features like framebuffer fetch? Perhaps we can detect that instead of tuning flags? Not sure. I bet this is fine too.

Comment (jonahwilliams) [reply impeller/tools/impeller.gni:681]: It interferes with framebuffer fetch specifically. I think there is a way to make that work but I haven't found it yet. I was planning to handle this as a follow up.

Comment (jonahwilliams) [reply impeller/tools/impeller.gni:681]: There's not really a great way to detect it automatically, since we only really get the chance to inspect the shader after its in a SPIR-V ast and at that point the validation that rejects the f16sampler2d has already run.

**Comment by jonahwilliams** [comment]
Friendly ping:) @dnfield @chinmaygarde

File: impeller/compiler/switches.h [hunk 2]
```diff
@@ -36,6 +36,7 @@ struct Switches {
   uint32_t gles_language_version;
   std::string metal_version;
   std::string entry_point;
+  bool use_half_textures;
```
Comment (chinmaygarde) [inline impeller/compiler/switches.h:39]: Default initialize all members. This and (if you are in the mood for drive-by changes) the other members in the struct.

**Review by chinmaygarde — APPROVED** [review]

File: impeller/compiler/switches.h (hunk 2 above)
Comment (jonahwilliams) [reply impeller/compiler/switches.h:39]: Done
